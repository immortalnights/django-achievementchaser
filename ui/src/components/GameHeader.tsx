import { Box, Grid, Typography } from "@mui/material"
import BorderedImage from "./BorderedImage"
import ExternalLink from "./ExternalLink"
import Link from "./Link"
import GameOwners from "./GameOwners"

const GameHeader = ({ game }: { game: Game }) => {
    return (
        <>
            <Box sx={{ display: "flex", marginBottom: "0.25em" }}>
                <Link to={`/game/${game.id}`}>{game.name}</Link>
                <Box
                    sx={{
                        display: "flex",
                        paddingX: 1,
                        alignItems: "flex-end",
                    }}
                >
                    <ExternalLink
                        href={`https://store.steampowered.com/app/${
                            game.id
                        }/${encodeURIComponent(game.name ?? "")}`}
                        title="Steam Profile"
                    />
                </Box>
            </Box>
            <Grid container>
                <Grid paddingRight="1em">
                    <BorderedImage
                        src={`https://media.steampowered.com/steam/apps/${game.id}/header.jpg`}
                    />
                </Grid>
                <Grid flexGrow={1}>
                    <Box
                        display="flex"
                        gap={2}
                        borderBottom="1px solid lightgray"
                    >
                        <Box>
                            <Typography
                                variant="subtitle1"
                                textTransform="uppercase"
                            >
                                Achievements
                            </Typography>
                            <Typography variant="body1">
                                {game.achievementCount ?? "?"}
                            </Typography>
                        </Box>
                    </Box>
                    <GameOwners game={game} />
                </Grid>
            </Grid>
        </>
    )
}

export default GameHeader
