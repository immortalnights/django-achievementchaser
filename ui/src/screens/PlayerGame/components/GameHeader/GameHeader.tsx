import { Stack } from "@mui/material"
import GameTitle from "@/components/GameTitle"
import GameIcon from "@/components/GameIcon"
import GameDifficulty from "@/components/GameDifficulty"
import AchievementControls from "../AchievementControls"
import Details from "../Details"

const GameHeader = ({
    game,
    owner,
    compare,
}: {
    game: Game
    owner?: Omit<PlayerOwnedGame, "game">
    compare: boolean
}) => {
    const gameAchievementCount = game.achievements?.length ?? 0

    return (
        <>
            <GameTitle game={game} />
            <Stack
                direction="row"
                justifyContent="space-between"
                alignItems="center"
                useFlexGap
                spacing={2}
            >
                <GameIcon {...game} />
                {gameAchievementCount > 0 && (
                    <>
                        <GameDifficulty
                            difficulty={game.difficultyPercentage}
                        />
                        <div style={{ width: "2em" }} />
                    </>
                )}

                {owner && (
                    <>
                        {!compare ? (
                            <Details game={game} owner={owner} />
                        ) : (
                            <div style={{ flexGrow: 1 }} />
                        )}

                        {gameAchievementCount > 0 && (
                            <AchievementControls game={game} owner={owner} />
                        )}
                    </>
                )}
            </Stack>
        </>
    )
}

export default GameHeader
