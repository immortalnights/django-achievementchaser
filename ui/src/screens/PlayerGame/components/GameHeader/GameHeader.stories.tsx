import type { Meta, StoryObj } from "@storybook/react"
import { useState } from "react"
import { MemoryRouter } from "react-router-dom"
import {
    antenna,
    antennaMultipleOwners,
    gameWithoutAchievements,
} from "@test/data"
import AchievementDisplayContext, {
    AchievementDisplayContextValue,
} from "../../context/AchievementDisplayContext"
import GameHeader from "."

const meta = {
    title: "Game Header 2",
    component: GameHeader,
    parameters: {
        layout: "centered",
    },
    decorators: [
        (Story) => (
            <MemoryRouter>
                <Story />
            </MemoryRouter>
        ),
    ],
    tags: ["autodocs"],
    argTypes: {},
} satisfies Meta<typeof GameHeader>

export default meta
type Story = StoryObj<typeof meta>

/**
 * Game with achievements, but no owner.
 * Displays title, icon, difficulty and search.
 */
export const WithAchievementsNotOwned: Story = {
    args: {
        game: antenna,
        owner: undefined,
        compare: false,
    },
}

/**
 * Game with achievements and single owner.
 * Displays title, icon, difficultly, owner stats, search and filtering.
 */
export const WithAchievementsOwned: Story = {
    args: {
        game: antenna,
        owner: antenna.owners.edges[0].node,
        compare: false,
    },
}

/**
 * Game with achievements and multiple owners.
 * Displays title, icon, difficulty, owner stats, search, compare and filtering.
 */
export const WithAchievementsCompare: Story = {
    args: {
        game: antennaMultipleOwners,
        owner: antennaMultipleOwners.owners.edges[0].node,
        compare: false,
    },
}

/**
 * Game with achievement comparison.
 * Displays title, icon, search, clear compare and filtering.
 */
export const Compare: Story = {
    decorators: [
        (Story) => {
            const [comparePlayer, setComparePlayer] = useState<
                string | undefined
            >("76561198013854782")

            const contextValue: AchievementDisplayContextValue = {
                filter: "",
                // eslint-disable-next-line @typescript-eslint/no-empty-function
                setFilter: () => {},
                otherPlayer: comparePlayer,
                setOtherPlayer: (value: string | undefined) =>
                    setComparePlayer(value),
            }

            return (
                <AchievementDisplayContext.Provider value={contextValue}>
                    <Story />
                </AchievementDisplayContext.Provider>
            )
        },
    ],
    args: {
        game: antennaMultipleOwners,
        owner: antennaMultipleOwners.owners.edges[0].node,
        compare: true,
    },
}

/**
 * Game without achievements or owner.
 * Displays title and icon.
 */
export const WithoutAchievementsNotOwned: Story = {
    args: {
        game: gameWithoutAchievements,
        owner: undefined,
        compare: false,
    },
}

/**
 * Game without achievements with owner.
 * Displays title, icon and owner stats.
 */
export const WithoutAchievementsOwned: Story = {
    args: {
        game: gameWithoutAchievements,
        owner: antenna.owners.edges[0].node,
        compare: false,
    },
}
