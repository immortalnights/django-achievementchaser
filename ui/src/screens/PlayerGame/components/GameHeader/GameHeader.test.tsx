import renderer from "react-test-renderer"
import { expect, test } from "vitest"
import GameHeader from "."
import { antenna } from "@test/data"
import { MemoryRouter } from "react-router-dom"

test("snapshot", () => {
    const component = renderer.create(
        <MemoryRouter>
            <GameHeader game={antenna} />
        </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
})
