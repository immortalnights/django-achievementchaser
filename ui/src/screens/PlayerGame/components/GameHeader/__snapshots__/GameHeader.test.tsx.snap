// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-mwah2a{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin:0 0 0.25em;-webkit-align-items:flex-end;-webkit-box-align:flex-end;-ms-flex-align:flex-end;align-items:flex-end;gap:8px;}",
      }
    }
    data-emotion="css mwah2a"
  />,
  <div
    className="MuiBox-root css-mwah2a"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-13v0ye7-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;}",
        }
      }
      data-emotion="css 13v0ye7-MuiLink-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-15jo3ss-MuiTypography-root-MuiLink-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1.5rem;line-height:1.334;letter-spacing:0em;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;}",
        }
      }
      data-emotion="css 15jo3ss-MuiTypography-root-MuiLink-root"
    />
    <a
      className="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineNone css-15jo3ss-MuiTypography-root-MuiLink-root"
      data-discover="true"
      href="/Game/443580"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      style={{}}
    >
      Antenna
    </a>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-o7e25t-MuiLink-root{-webkit-text-decoration:underline;text-decoration:underline;text-decoration-color:var(--Link-underlineColor);--Link-underlineColor:rgba(25, 118, 210, 0.4);}.css-o7e25t-MuiLink-root:hover{text-decoration-color:inherit;}",
        }
      }
      data-emotion="css o7e25t-MuiLink-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1orsvah-MuiTypography-root-MuiLink-root{margin:0;font:inherit;line-height:inherit;letter-spacing:inherit;color:#1976d2;-webkit-text-decoration:underline;text-decoration:underline;text-decoration-color:var(--Link-underlineColor);--Link-underlineColor:rgba(25, 118, 210, 0.4);}.css-1orsvah-MuiTypography-root-MuiLink-root:hover{text-decoration-color:inherit;}",
        }
      }
      data-emotion="css 1orsvah-MuiTypography-root-MuiLink-root"
    />
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
      href="http://store.steampowered.com/app/443580"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      style={{}}
      target="_blank"
      title="Steam Game"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-120dh41-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;fill:currentColor;font-size:1.25rem;}",
          }
        }
        data-emotion="css 120dh41-MuiSvgIcon-root"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
        />
      </svg>
    </a>
  </div>,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1oh6lrx-MuiStack-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:16px;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
      }
    }
    data-emotion="css 1oh6lrx-MuiStack-root"
  />,
  <div
    className="MuiStack-root css-1oh6lrx-MuiStack-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      alt="Antenna"
      className="css-mt3xud"
      id="443580"
      name="Antenna"
      src="https://media.steampowered.com/steam/apps/443580/capsule_184x69.jpg"
    />
    <div
      style={
        {
          "minWidth": "85px",
        }
      }
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1s1s9f7-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.75;letter-spacing:0.00938em;text-transform:uppercase;}",
          }
        }
        data-emotion="css 1s1s9f7-MuiTypography-root"
      />
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 css-1s1s9f7-MuiTypography-root"
        style={{}}
      >
        Difficulty
      </h6>
      18.50
      %
    </div>
    <div
      style={
        {
          "flexGrow": 0.5,
        }
      }
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1fikxhz{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:end;-ms-flex-pack:end;-webkit-justify-content:flex-end;justify-content:flex-end;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        }
      }
      data-emotion="css 1fikxhz"
    />
    <div
      className="MuiBox-root css-1fikxhz"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1orzy9s{position:relative;border-radius:4px;background-color:rgba(255, 255, 255, 0.15);margin-left:0;width:100%;}.css-1orzy9s:hover{background-color:rgba(255, 255, 255, 0.25);}@media (min-width:600px){.css-1orzy9s{margin-left:8px;width:auto;}}",
          }
        }
        data-emotion="css 1orzy9s"
      />
      <div
        className="css-1orzy9s"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1iz4iln{padding:0px 16px;height:100%;position:absolute;pointer-events:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
            }
          }
          data-emotion="css 1iz4iln"
        />
        <div
          className="css-1iz4iln"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1umw9bq-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;fill:currentColor;font-size:1.5rem;}",
              }
            }
            data-emotion="css 1umw9bq-MuiSvgIcon-root"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
            />
          </svg>
        </div>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-12k6a2e{color:inherit;}.css-12k6a2e .MuiInputBase-input{padding:8px 8px 8px 0px;padding-left:calc(1em + 32px);-webkit-transition:width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;width:100%;}@media (min-width:600px){.css-12k6a2e .MuiInputBase-input{width:20ch;}}",
            }
          }
          data-emotion="css 12k6a2e"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": "@-webkit-keyframes mui-auto-fill{from{display:block;}}@keyframes mui-auto-fill{from{display:block;}}@-webkit-keyframes mui-auto-fill-cancel{from{display:block;}}@keyframes mui-auto-fill-cancel{from{display:block;}}",
            }
          }
          data-emotion="css-global 1prfaxn"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1g6b8oh-MuiInputBase-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.4375em;letter-spacing:0.00938em;color:rgba(0, 0, 0, 0.87);box-sizing:border-box;position:relative;cursor:text;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:inherit;}.css-1g6b8oh-MuiInputBase-root.Mui-disabled{color:rgba(0, 0, 0, 0.38);cursor:default;}.css-1g6b8oh-MuiInputBase-root .MuiInputBase-input{padding:8px 8px 8px 0px;padding-left:calc(1em + 32px);-webkit-transition:width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;width:100%;}@media (min-width:600px){.css-1g6b8oh-MuiInputBase-root .MuiInputBase-input{width:20ch;}}",
            }
          }
          data-emotion="css 1g6b8oh-MuiInputBase-root"
        />
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary css-1g6b8oh-MuiInputBase-root"
          onClick={[Function]}
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-yimnyd-MuiInputBase-input{font:inherit;letter-spacing:inherit;color:currentColor;padding:4px 0 5px;border:0;box-sizing:content-box;background:none;height:1.4375em;margin:0;-webkit-tap-highlight-color:transparent;display:block;min-width:0;width:100%;-webkit-animation-name:mui-auto-fill-cancel;animation-name:mui-auto-fill-cancel;-webkit-animation-duration:10ms;animation-duration:10ms;}.css-yimnyd-MuiInputBase-input::-webkit-input-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-yimnyd-MuiInputBase-input::-moz-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-yimnyd-MuiInputBase-input::-ms-input-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-yimnyd-MuiInputBase-input:focus{outline:0;}.css-yimnyd-MuiInputBase-input:invalid{box-shadow:none;}.css-yimnyd-MuiInputBase-input::-webkit-search-decoration{-webkit-appearance:none;}label[data-shrink=false]+.MuiInputBase-formControl .css-yimnyd-MuiInputBase-input::-webkit-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-yimnyd-MuiInputBase-input::-moz-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-yimnyd-MuiInputBase-input::-ms-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-yimnyd-MuiInputBase-input:focus::-webkit-input-placeholder{opacity:0.42;}label[data-shrink=false]+.MuiInputBase-formControl .css-yimnyd-MuiInputBase-input:focus::-moz-placeholder{opacity:0.42;}label[data-shrink=false]+.MuiInputBase-formControl .css-yimnyd-MuiInputBase-input:focus::-ms-input-placeholder{opacity:0.42;}.css-yimnyd-MuiInputBase-input.Mui-disabled{opacity:1;-webkit-text-fill-color:rgba(0, 0, 0, 0.38);}.css-yimnyd-MuiInputBase-input:-webkit-autofill{-webkit-animation-duration:5000s;animation-duration:5000s;-webkit-animation-name:mui-auto-fill;animation-name:mui-auto-fill;}",
              }
            }
            data-emotion="css yimnyd-MuiInputBase-input"
          />
          <input
            aria-label="filter achievements"
            className="MuiInputBase-input css-yimnyd-MuiInputBase-input"
            name="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
